runtime/testy/stateManager.py

```diff
@@ -91,21 +98,40 @@ def setValue(self, value, keys):
     currDict = self.state
     i = 0
     try:
+      path = []
       for key in enumerate(keys[:-1]):
         i = key[0]
-        currDict = currDict[key[1]]
+        path.append(currDict[key[1]])
+        currDict = currDict[key[1]][0]
       if len(keys) > 1:
         i += 1
       if keys[i] not in currDict:
         raise Exception
-      currDict[keys[i]] = value
+      path.append(currDict[keys[i]])
+      currDict[keys[i]][0] = value
+      currTime = time.time();
+      for item in path:
```

Comment:
Same comment as above. We can accomplish the same result in a single pass by updating timestamps as we search into the dictionary.